truckersmp_cli/__init__.py

```diff
@@ -492,21 +492,46 @@ def start_with_wine():
     if args.activate_native_d3dcompiler_47:
         activate_native_d3dcompiler_47(args.prefixdir, wine)
 
-    print("""
-    ###################################################################
-    #                                                                 #
-    #  Please check wine steam is running or the launcher won't work  #
-    #                                                                 #
-    ###################################################################
-
-    Press enter if you are good to go: """, end="")
-    sys.stdin.readline()
-
     env = os.environ.copy()
     env["WINEDEBUG"] = "-all"
     env["WINEARCH"] = "win64"
     env["WINEPREFIX"] = args.prefixdir
     env["WINEDLLOVERRIDES"] = "d3d11=;dxgi=" if not args.enable_d3d11 else ""
+
+    # make sure that Steam is running
+    steam_is_running = False
+    argv = (wine, "winedbg", "--command", "info process")
+    try:
+        output = subproc.check_output(argv, env=env, stderr=subproc.DEVNULL)
+        for line in output.decode("utf-8").splitlines():
+            line = line[:-1]  # strip last "'" for rindex()
+            try:
+                exename = line[line.rindex("'") + 1:]
+            except ValueError:
+                continue
+            if exename.lower().endswith("steam.exe"):
+                steam_is_running = True
+                break
+    except subproc.CalledProcessError as e:
+        sys.exit("Failed to get Wine process list: " + e.output.decode("utf-8"))
+    if not steam_is_running:
+        print("Waiting for Steam to be running...", end="")
+        sys.stdout.flush()
+        loginusers_vdf = os.path.join(args.wine_steam_dir, "config/loginusers.vdf")
+        try:
+            old_timestamp = os.stat(loginusers_vdf).st_mtime
+        except OSError:
+            old_timestamp = 0
+        while True:
+            try:
+                timestamp = os.stat(loginusers_vdf).st_mtime
+                if timestamp > old_timestamp:
+                    print("\r{}".format(" " * 70))
+                    break
+            except OSError:
+                pass
+            time.sleep(1)
```

Comment:
We're now basically doing the same thing as above for proton with a different path. Can't we combine them into a unified function or something similar?